bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/autoupdate/internal/AutoUpdateProviderDelegate.java

```diff
@@ -20,26 +37,185 @@
  *
  * @author Kai Kreuzer - Initial contribution and API
  */
-public class AutoUpdateProviderDelegate implements AutoUpdateBindingConfigProvider {
+@NonNullByDefault
+@Component(service = MetadataProvider.class)
+public class AutoUpdateProviderDelegate
+        implements MetadataProvider, RegistryChangeListener<Item>, BindingChangeListener {
+
+    private static final String AUTOUPDATE_KEY = "autoupdate";
 
     private Set<org.openhab.core.autoupdate.AutoUpdateBindingProvider> providers = new CopyOnWriteArraySet<>();
+    private Set<ProviderChangeListener<Metadata>> listeners = new CopyOnWriteArraySet<>();
+    private Set<String> itemUpdateVetos = new HashSet<>();
+    private boolean started = false;
+
+    private @NonNullByDefault({}) ItemRegistry itemRegistry;
+
+    @Activate
+    protected void activate() {
+        refreshItemUpdateVetos();
+        started = true;
+        itemRegistry.addRegistryChangeListener(this);
+        for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+            provider.addBindingChangeListener(this);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+            provider.removeBindingChangeListener(this);
+        }
+        itemRegistry.removeRegistryChangeListener(this);
+        started = false;
+    }
 
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE)
     public void addAutoUpdateBindingProvider(org.openhab.core.autoupdate.AutoUpdateBindingProvider provider) {
```

Comment:
It seems we can import these now, given the one from ESH is not imported anymore.